veems/media/api_views.py

```diff
@@ -52,12 +53,27 @@ def upload_complete(request, upload_id):
     return Response({}, status=OK)
 
 
-@api_view(['GET'])
-@permission_classes([AllowAny])
-def video(request, video_id):
-    video = models.Video.objects.get(id=video_id)
-    data = serializers.VideoSerializer(video).data
-    return Response(data, status=OK)
+class VideoDetailAPIView(APIView):
+    permission_classes = [IsAuthenticatedOrReadOnly]
+
+    def get(self, request, video_id, format=None):
+        video = models.Video.objects.get(id=video_id)
+        data = serializers.VideoSerializer(video).data
+        return Response(data, status=OK)
+
+    def put(self, request, video_id, format=None):
+        video = models.Video.objects.get(
+            id=video_id, channel__user_id=request.user.id
+        )
+        serializer = serializers.VideoSerializer(
+            video, data=request.data, partial=True
+        )
+        if serializer.is_valid():
+            video = serializer.save()
+            serializer = serializers.VideoSerializer(video)
+            return Response(serializer.data)
+        else:
+            return Response({'detail': 'Invalid payload'}, status=BAD_REQUEST)
```

Comment:
```suggestion
serializer.is_valid(raise_exception=True)
video = serializer.save()
serializer = serializers.VideoSerializer(video)
return Response(serializer.data)
```

passing in `raise_exception` returns 400 with the validation errors in the body

is the serializer being instantiated again because the partial one only contains a subset of the fields?

Reply:
:+1: will fix that this eve. And yes, I think passing partial=True allows you to just pass the model fields you'd like to be updated, rather than all mode fields (https://www.django-rest-framework.org/api-guide/serializers/#partial-updates)